src/miral/input_device_config.cpp

```diff
@@ -62,56 +78,187 @@ char const* const touchpad_click_mode_opt= "touchpad-click-mode";
+        {
+            return mir_touchpad_scroll_mode_button_down_scroll;
+        }
+        else
+        {
+            throw mir::AbnormalExit{std::format("Unrecognised option for {}: {}", touchpad_scroll_mode_opt, *opt_val)};
+        }
+    }
+    else
+    {
+        return std::nullopt;
+    }
+}
+
+auto to_acceleration_profile(std::optional<std::string> const& opt_val)-> std::optional<MirPointerAcceleration>
+{
+    if (opt_val)
+    {
+        if (*opt_val == acceleration_none)
+        {
+            return mir_pointer_acceleration_none;
+        }
+        else if (*opt_val == acceleration_adaptive)
+        {
+            return mir_pointer_acceleration_adaptive;
+        }
+        else
+        {
+            throw mir::AbnormalExit{std::format("Unrecognised option for cursor-acceleration: {}", *opt_val)};
+        }
+    }
+    else
+    {
+        return std::nullopt;
+    }
+}
+
+auto convert_to_click_mode(std::optional<std::string> const& opt_val)
+-> std::optional<MirTouchpadClickMode>
+{
+    if (opt_val)
+    {
+        auto val = *opt_val;
+        if (val == touchpad_click_mode_none)
+        {
+            return mir_touchpad_click_mode_none;
+        }
+        else if (val == touchpad_click_mode_clickfinger)
+        {
+            return mir_touchpad_click_mode_finger_count;
+        }
+        else if (val == touchpad_click_mode_area)
+        {
+            return mir_touchpad_click_mode_area_to_click;
+        }
+        else
+        {
+            throw mir::AbnormalExit{std::format("Unrecognised option for {}: {}", touchpad_click_mode_opt, *opt_val)};
+        }
+    }
+    else
+    {
+        return std::nullopt;
+    }
+}
+
+auto to_handedness(std::optional<std::string> const& opt_val)-> std::optional<MirPointerHandedness>
+{
+    if (opt_val)
+    {
+        if (*opt_val == right)
+        {
+            return mir_pointer_handedness_right;
+        }
+        else if (*opt_val == left)
+        {
+            return mir_pointer_handedness_left;
+        }
+        else
+        {
+            throw mir::AbnormalExit{std::format("Unrecognised option for handedness: {}", *opt_val)};
+        }
+    }
+    else
+    {
+        return std::nullopt;
+    }
+}
 }
 
 void miral::add_input_device_configuration_options_to(mir::Server& server)
 {
-    server.add_configuration_option(mouse_acceleration_opt,
+    server.add_configuration_option(mouse_handedness_opt, std::format("Select mouse laterality: [{}, {}]", right, left),
+    mir::OptionType::string);
+    server.add_configuration_option(mouse_cursor_acceleration_opt,
                                     "Select acceleration profile for mice and trackballs [none, adaptive]",
                                     mir::OptionType::string);
     server.add_configuration_option(mouse_cursor_acceleration_bias_opt,
                                     "Constant factor (+1) to velocity or bias to the acceleration curve within the range [-1.0, 1.0] for mice",
```

Comment:
I know it's unchanged here, but while we're at it, what does (+1) refer to here?

It doesn't seem to be the default value. What is it?

Reply:
Git blames me for the text, but I don't know either. (FWIW I think Andreas wrote it)